### PR TITLE
Settings components

### PR DESCRIPTION
**Pull request by jequeen**
- Created new settings screen for tablet and phone in both landscape and portrait

### CODE REVIEW

File: app/src/main/java/com/willowtree/vocable/settings/SettingsFragment.kt [hunk 1]
```diff
@@ -135,13 +145,14 @@ class SettingsFragment : BaseFragment() {
     private fun subscribeToViewModel() {
         viewModel.headTrackingEnabled.observe(viewLifecycleOwner, Observer {
             it?.let {
-                binding?.headTrackingSwitch?.isChecked = it
+                //This setting has been removed from the main settings page
+                //binding?.headTrackingSwitch?.isChecked = it
```
Comment (nicholas-cook) [inline app/src/main/java/com/willowtree/vocable/settings/SettingsFragment.kt]: Make sure we remove these comments when this logic gets moved.

File: app/src/main/java/com/willowtree/vocable/settings/SettingsFragment.kt [hunk 2]
```diff
@@ -20,17 +22,35 @@ class SettingsFragment : BaseFragment() {
...
+
+        numColumns = resources.getInteger(R.integer.settings_options_columns)
+
+        (binding?.settingsOptionsContainer?.root as? GridLayout)?.children?.forEachIndexed { index, child ->
+            if (index % numColumns == numColumns - 1) {
+                child.layoutParams = (child.layoutParams as GridLayout.LayoutParams).apply {
+                    marginEnd = 0
+                }
+            }
+            if (index > SETTINGS_OPTION_COUNT - numColumns) {
```
[the remark this answers is not in the document]
Comment (jequeen) [reply app/src/main/java/com/willowtree/vocable/settings/SettingsFragment.kt]: I checked each of the layouts and it works as intended.

File: app/src/main/res/layout-sw600dp-land/fragment_settings.xml [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        app:layout_constraintBottom_toBottomOf="@+id/settings_close_button"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="@+id/settings_close_button"
+        app:layout_constraintVertical_chainStyle="packed" />
+
+    <include
+        android:id="@+id/settings_options_container"
+        layout="@layout/settings_options_layout"
+        android:layout_width="match_parent"
```
Comment (MattCorbett488) [inline app/src/main/res/layout-sw600dp-land/fragment_settings.xml]: It's recommended that, whenever you wanna use `match_parent` in a `ConstraintLayout`, that we set the start/end to `parent` and give a width of 0dp (MATCH_CONSTRAINT).

To prove I'm not lying, it's in the [docs](https://developer.android.com/reference/android/support/constraint/ConstraintLayout#wrap_content-:-enforcing-constraints--added-in-1.1-) as a note

File: app/src/main/res/layout/settings_option_buttons.xml [hunk 4]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="utf-8"?>
+<com.willowtree.vocable.customviews.ActionButton xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    tools:layout_width="match_parent"
+    android:layout_width="0dp"
+    android:layout_height="48dp"
```
Comment (MattCorbett488) [inline app/src/main/res/layout/settings_option_buttons.xml]: Few hardcoded values here - fine if they're not going to be reused elsewhere

File: app/src/main/res/layout/settings_options_layout.xml [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+    android:layout_width="match_parent"
+    android:layout_height="0dp"
+    android:columnCount="@integer/settings_options_columns"
+    android:rowCount="@integer/settings_options_rows">
+
+    <com.willowtree.vocable.customviews.SettingsButton
+        android:id="@+id/edit_sayings_button"
+        android:layout_width="0dp"
+        android:layout_height="0dp"
+        android:layout_rowWeight="1"
```
Comment (MattCorbett488) [inline app/src/main/res/layout/settings_options_layout.xml]: You should be able to extract a lot of these properties to styles (it looks like the `android:text` is one of the only unique attributes on most of these buttons?

**Pull request merged**